tokio/tests/io_copy_bidirectional.rs

```diff
@@ -138,3 +138,28 @@ async fn immediate_exit_on_read_error() {
 
     assert!(copy_bidirectional(&mut a, &mut b).await.is_err());
 }
+
+#[tokio::test]
+async fn copy_bidirectional_is_cooperative() {
```

Comment:
It seems the copy_bidirectional is already cooperative before I make the CopyBuffer cooperative. I'm not sure why. Could I be using the wrong case to test it?

Reply:
You should probably change this test to use something different than `tokio_test::io::Builder` to fix that.

Reply:
I tried to use `symmetric`, but it also didn't work.